test/e2e/route_advertisements.go

```diff
@@ -178,37 +180,217 @@ var _ = ginkgo.Describe("BGP: Pod to external server when default podNetwork is
-					"and expecting to receive the same payload", serverContainerIP))
-				cmd := fmt.Sprintf("curl --max-time 10 -g -q -s http://%s/clientip",
-					net.JoinHostPort(serverContainerIP, "8080"),
-				)
-				framework.Logf("Testing pod to external traffic with command %q", cmd)
-				stdout, err := e2epodoutput.RunHostCmdWithRetries(
-					clientPod.Namespace,
-					clientPod.Name,
-					cmd,
-					framework.Poll,
-					60*time.Second)
-				framework.ExpectNoError(err, fmt.Sprintf("Testing pod to external traffic failed: %v", err))
-				expectedPodIP := podv4IP
-				if isIPv6Supported(f.ClientSet) && utilnet.IsIPv6String(serverContainerIP) {
-					expectedPodIP = podv6IP
-					// For IPv6 addresses, need to handle the brackets in the output
-					outputIP := strings.TrimPrefix(strings.Split(stdout, "]:")[0], "[")
-					gomega.Expect(outputIP).To(gomega.Equal(expectedPodIP),
-						fmt.Sprintf("Testing pod %s to external traffic failed while analysing output %v", echoClientPodName, stdout))
-				} else {
-					// Original IPv4 handling
-					gomega.Expect(strings.Split(stdout, ":")[0]).To(gomega.Equal(expectedPodIP),
-						fmt.Sprintf("Testing pod %s to external traffic failed while analysing output %v", echoClientPodName, stdout))
+			if podv4IP != "" {
+				clientPodIPs = append(clientPodIPs, podv4IP)
+			}
+			if podv6IP != "" {
+				clientPodIPs = append(clientPodIPs, podv6IP)
+			}
+
+			clientNode, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), clientPod.Spec.NodeName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			// Get the nodeIPs for the node where the client pod is running
+			var nodeIPv4, nodeIPv6 string
+			if isIPv4Supported(f.ClientSet) {
+				v4Addrs := e2enode.GetAddressesByTypeAndFamily(clientNode, corev1.NodeInternalIP, corev1.IPv4Protocol)
+				if len(v4Addrs) > 0 {
+					nodeIPv4 = v4Addrs[0]
+				}
+			}
+			if isIPv6Supported(f.ClientSet) {
+				v6Addrs := e2enode.GetAddressesByTypeAndFamily(clientNode, corev1.NodeInternalIP, corev1.IPv6Protocol)
+				if len(v6Addrs) > 0 {
+					nodeIPv6 = v6Addrs[0]
 				}
 			}
+			if nodeIPv4 != "" {
+				clientPodNodeIPs = append(clientPodNodeIPs, nodeIPv4)
+			}
+			if nodeIPv6 != "" {
+				clientPodNodeIPs = append(clientPodNodeIPs, nodeIPv6)
+			}
+			framework.Logf("clientPodNodeIPs: %v", clientPodNodeIPs)
+
+			ginkgo.By("Creating a host networked pod on a second node")
+			// Add labels so the service can select this pod
+			podLabels := map[string]string{
+				"app": "internal-server-pod",
+			}
+
+			// Create host networked pod on another cluster node using createPod function
+			hostNetworkedPod, err = createPod(f, "internal-server-pod", nodes.Items[2].Name, f.Namespace.Name,
+				[]string{"/agnhost", "netexec", "--http-port=8080"}, podLabels, func(p *corev1.Pod) {
+					// Set host networking
+					p.Spec.HostNetwork = true
+
+					// Add required security context to comply with PodSecurity "restricted" policy
```

Comment:
I know it's required by OCP clusters. But is that necessary for the u/s cluster?

Reply:
If I do not add security context, I would get:
 I0904 16:46:16.714950 1651793 warnings.go:110] "Warning: would violate PodSecurity \"restricted:latest\": host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container \"internal-server-pod-container\" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container \"internal-server-pod-container\" must set securityContext.capabilities.drop=[\"ALL\"]), runAsNonRoot != true (pod or container \"internal-server-pod-container\" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container \"internal-server-pod-container\" must set securityContext.seccompProfile.type to \"RuntimeDefault\" or \"Localhost\")"

Reply:
I am going to keep it